### PR TITLE
test: add fri opening benchmark

### PR DESCRIPTION
**Pull request by batzor**
# Description

This PR optimizes some part of FRI and adds opening phase to the benchmark.

### CODE REVIEW

File: tachyon/crypto/challenger/challenger.h [hunk 1]
```diff
@@ -111,7 +111,8 @@ class Challenger {
       return v != std::numeric_limits<uint32_t>::max();
     });
     CHECK(it != ret.end());
-    CheckWitness(bits, Field(*it));
+    Derived* derived = static_cast<Derived*>(this);
+    CHECK(derived->CheckWitness(bits, Field(*it)));
```
Comment (chokobole) [inline tachyon/crypto/challenger/challenger.h:121]: Please add `[[nodiscard]]` to `CheckWitness()`, too!

```suggestion
    CHECK(CheckWitness(bits, Field(*it)));
```

**Comment by chokobole** [comment]
[chore(benchmark): run cargo fmt](https://github.com/kroma-network/tachyon/pull/533/commits/2df0bce5480d4b59133d71a8d1e5630564a623fe) should be typed as a style.

File: tachyon/crypto/commitments/fri/two_adic_fri.h [hunk 2]
```diff
@@ -74,14 +74,11 @@ class TwoAdicFRI {
...
-          math::RowMajorMatrix<F> ret = coset.domain()->CosetLDEBatch(
-              mat, fri_.log_blowup,
+          return coset.domain()->CosetLDEBatch(
+              matrices[i], fri_.log_blowup,
               F::FromMontgomery(F::Config::kSubgroupGenerator) *
-                  coset.domain()->offset_inv());
-          ReverseMatrixIndexBits(ret);
-          return ret;
+                  coset.domain()->offset_inv(),
+              /*reverse_at_last=*/false);
```
Comment (chokobole) [inline tachyon/crypto/commitments/fri/two_adic_fri.h:83]: Could you apply this to `CosetLDEBatch()` in `two_adic_fri_impl.h`, too?

**Review by chokobole — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by GideokKim — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**